src/perception/PoseEstimatorModule.cpp

```diff
@@ -63,25 +62,31 @@ bool PoseEstimatorModule::detectObjects(
 
   for (const auto& marker_transform : marker_message->markers)
   {
+    // TODO: Add DELETE_ALL Functionality
+    if (marker_transform.action == visualization_msgs::Marker::DELETEALL) {
+      dtwarn << "[PoseEstimatorModule::detectObjects] We cannot currently handle DELETE_ALL markers." << std::endl;
+      continue;
+    }
     const auto& marker_stamp = marker_transform.header.stamp;
-    const auto& obj_key = marker_transform.text;
-    const auto& obj_id = marker_transform.ns;
+    const auto& obj_id = marker_transform.id;
+    const auto& obj_ns = marker_transform.ns;
     const auto& detection_frame = marker_transform.header.frame_id;
     if (!timestamp.isValid() || marker_stamp < timestamp)
     {
       continue;
     }
 
-    // If obj_key is "None",
-    // remove a skeleton with obj_id from env
-    // and move to next marker
-    dart::dynamics::SkeletonPtr env_skeleton = env->getSkeleton(obj_id);
-    if (!obj_key.compare("None"))
-    {
-      if (env_skeleton != nullptr)
-      {
-        env->removeSkeleton(env_skeleton);
-      }
+    const std::string obj_uid = obj_ns + "_" + std::to_string(obj_id);
```

Comment:
We should mention in the header of this file that this is how we assign uids to objects.